lib/castle/core/process_response.rb

```diff
@@ -36,8 +37,25 @@ def verify!(response)
 
           raise Castle::InternalServerError if response.code.to_i.between?(500, 599)
 
+          raise error422(response) if response.code.to_i == 422
+
           error = RESPONSE_ERRORS.fetch(response.code.to_i, Castle::ApiError)
-          raise error, response[:message]
+
+          raise error
+        end
+
+        def error422(response)
+          if response.body
+            begin
+              parsed_body = JSON.parse(response.body, symbolize_names: true)
+              if parsed_body.is_a?(Hash) && parsed_body[:type] == INVALID_REQUEST_TOKEN
+                return Castle::InvalidRequestTokenError
+              end
+            rescue JSON::ParserError
+            end
+          end
+
+          Castle::InvalidParametersError
```

Comment:
Would be great to add a proper error message to these exceptions. Or is this going to be a different PR? :)

Reply:
👍  good point